src/BCPlayer.js

```diff
@@ -1,91 +1,148 @@
-import React, { Component } from 'react';
-import { StyleSheet, Platform } from 'react-native';
-import BrightcovePlayer from './BrightcovePlayer';
+import React, {Component} from 'react'
+import {Animated, BackHandler, Dimensions, Image, Platform, StatusBar, StyleSheet, Text} from 'react-native'
+import Orientation from 'react-native-orientation'
+import BrightcovePlayer from "./BrightcovePlayer";
+import PlayerEventTypes from "./PlayerEventTypes";
+
+const checkSource = (uri) => {
+	return typeof uri === 'string' ?
+		{source: {uri}} : {source: uri}
+}
 
-import Orientation from 'react-native-orientation';
+const Win = Dimensions.get('window')
+const backgroundColor = '#000'
 
-import PlayerEventTypes from './PlayerEventTypes';
+const styles = StyleSheet.create({
+	background: {
+		backgroundColor,
+		justifyContent: 'center',
+		alignItems: 'center',
+		zIndex: 98
+	},
+	fullScreen: {
+		...StyleSheet.absoluteFillObject
+	},
+	image: {
+		...StyleSheet.absoluteFillObject,
+		width: undefined,
+		height: undefined,
+		zIndex: 99
+	}
+})
 
 class BCPlayer extends Component {
-
 	constructor(props) {
-		super(props);
-
+		super(props)
 		this.state = {
-			orientation: null,
-			forcedOrientation: false,
-			percentageTracked: { Q1: false, Q2: false, Q3: false, Q4: false },
+			paused: !props.autoPlay,
+			fullScreen: false,
+			inlineHeight: Win.width * 0.5625,
+			loading: false,
+			currentTime: 0,
+			percentageTracked: {Q1: false, Q2: false, Q3: false, Q4: false},
 			mediainfo: null
 		}
-
-		this.orientationDidChange = this.orientationDidChange.bind(this);
-
-	}
-
-	componentWillMount() {
-		// The getOrientation method is async. It happens sometimes that
-		// you need the orientation at the moment the JS runtime starts running on device.
-		// `getInitialOrientation` returns directly because its a constant set at the
-		// beginning of the JS runtime.
-
-		const initial = Orientation.getInitialOrientation();
-		this.setState({ orientation: initial });
-		// Remember to remove listener
-		Orientation.removeOrientationListener(this.orientationDidChange);
+		this.animInline = new Animated.Value(Win.width * 0.5625)
+		this.animFullscreen = new Animated.Value(Win.width * 0.5625)
+		this.BackHandler = this.BackHandler.bind(this)
+		this.onRotated = this.onRotated.bind(this)
 	}
 
 	componentDidMount() {
-		Orientation.addOrientationListener(this.orientationDidChange);
+		Dimensions.addEventListener('change', this.onRotated)
+		BackHandler.addEventListener('hardwareBackPress', this.BackHandler)
 	}
 
-	onBeforeEnterFullscreen() {
+	componentWillUnmount() {
+		Dimensions.removeEventListener('change', this.onRotated)
+		BackHandler.removeEventListener('hardwareBackPress', this.BackHandler)
+	}
 
-		if (this.state.orientation === 'PORTRAIT') {
-			this.setState({ forcedOrientation: true });
-			Orientation.lockToLandscape();
+	onRotated({window: {width, height}}) {
+		if (this.props.inlineOnly) return
```

Comment:
@pankaj-pp  i noticed that most of the lines doesn't end with ";". 
I would highly recommend to do so, as a good practice in Javascript.

Reply:
@rafbc yes, I'll do that. Actually we don't use semicolons in our projects, so that's the reason i missed it.